SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -989,6 +989,43 @@ void USpatialNetDriver::ServerReplicateActors_ProcessPrioritizedActors(UNetConne
 	// SpatialGDK - Here Unreal would return the position of the last replicated actor in PriorityActors before the channel became saturated.
 	// In Spatial we use ActorReplicationRateLimit and EntityCreationRateLimit to limit replication so this return value is not relevant.
 }
+
+void USpatialNetDriver::ProcessRPC(AActor* Actor, UObject* SubObject, UFunction* Function, void* Parameters)
+{
+	// The RPC might have been called by an actor directly, or by a subobject on that actor
+	UObject* CallingObject = SubObject != nullptr ? SubObject : Actor;
+
+	if (IsServer())
+	{
+		// Creating channel to ensure that object will be resolvable
+		GetOrCreateSpatialActorChannel(CallingObject);
+	}
+
+	int ReliableRPCIndex = 0;
+	if (GetDefault<USpatialGDKSettings>()->bCheckRPCOrder)
+	{
+		if (Function->HasAnyFunctionFlags(FUNC_NetReliable) && !Function->HasAnyFunctionFlags(FUNC_NetMulticast))
+		{
+			ReliableRPCIndex = GetNextReliableRPCId(Actor, FunctionFlagsToRPCSchemaType(Function->FunctionFlags), CallingObject);
+		}
+	}
+
+	TSet<TWeakObjectPtr<const UObject>> UnresolvedObjects;
+	RPCPayload Payload = Sender->CreateRPCPayloadFromParams(CallingObject, Function, ReliableRPCIndex, Parameters, UnresolvedObjects);
+
+	if (UnresolvedObjects.Num() == 0)
+	{
+		FPendingRPCParamsPtr RPCParams = MakeShared<FPendingRPCParams>(CallingObject, Function, MoveTemp(Payload), ReliableRPCIndex);
+		Sender->QueueOutgoingRPC(RPCParams);
+		// TODO(Alex): Create jira ticket to remove queueing when not needed
+		Sender->SendOutgoingRPCs();
+	}
+	else
+	{
+		UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("The object %s is unresolved, RPC %s will be dropped"), *UnresolvedObjects.CreateIterator()->Get()->GetName(), *Function->GetName());
```

Comment:
When does this happen. It looks like we do it regardless of whether the RPC is reliable or not.

Reply:
That happens in all cases, yes. But as I understand, if the sender works as expected - it cannot have anything unresolved at that point?

Reply:
If we ignore the singleton case, yes.

Reply:
Resolving, since there's no pending suggestion.

Reply:
Discussed offline - not a problem in single server, but should document this for multiworker